packages/edit-site/src/components/routes/index.js

```diff
@@ -39,7 +39,14 @@ export function Routes( { children } ) {
 
 	useEffect( () => {
 		return history.listen( ( { location: updatedLocation } ) => {
-			setLocation( getLocationWithParams( updatedLocation ) );
+			setLocation( ( currentLocation ) => {
+				// Skip location update if query args are identical.
+				if ( currentLocation.search === updatedLocation.search ) {
```

Comment:
This only considers the query args, what if the url itself changes?

Reply:
As far as I can tell current links only update the query args. The `pathname` is always the same.

Example location:

```json
{
    "pathname": "/wp-admin/site-editor.php",
    "search": "?postType=wp_template_part&postId=twentytwentythree%2F%2Ffooter&sidebar=%2Ftemplate-parts",
    "hash": "",
    "state": null,
    "key": "xj26ne4s"
}
```

Reply:
Yeah, it's less than ideal that the router in the site editor is not generic enough because of the technical limitation back then (I wonder if any of them has changed?).

We can construct a url string based on the location object and compare them if we want to.

One thing that I totally forgot is whether we're still using the `state` key here? Maybe it's used in the focus mode for tracking the back button? Might be worth checking it out and see if it still works.